### PR TITLE
Dask Cluster Integration

### PR DESCRIPTION
**Pull request by DarshanSP19**
- (Optional) Connect to the `Dask` cluster before starting the tool.

Steps to Use
---
- Deploy a `Dask` cluster on Google `Kubernetes` Engine.
- Set environment Variable `DASK_CLUSTER={dask_cluster_name}` As a default xql-dask will be used as a `Dask` cluster name.
- Cluster will be automatically connected when the `CLI`.

### CODE REVIEW

File: xql/src/xql/apply.py [hunk 1]
```diff
@@ -432,28 +432,19 @@ def display_table_dataset_map(cmd: str) -> None:
...
-    """
-    if isinstance(result, xr.Dataset):
-        df = filter_records(convert_to_dataframe(result), query)
-        print(df)
-    else:
-        print(result)
-
 @timing
-def run_query(query: str) -> None:
+def run_query(query: str, connect_cluster = True) -> None:
```
Comment (mahrsee1997) [inline xql/src/xql/apply.py:436]: what is the use of 'connect_cluster' argument here ?

File: xql/README.md [hunk 2]
```diff
@@ -94,6 +94,28 @@ python xql/main.py
...
+    gcloud container clusters get-credentials {cluster_name} --region {cluster_region} --project {project}
+    ```
+3. Install `helm`. Follow [Helm | Installing Helm](https://helm.sh/docs/intro/install/). 
+4. Deploy Dask Cluster using below `helm` commands.
+    ```
+    helm repo add dask https://helm.dask.org/
+    helm repo update
+    helm install xql-dask dask/dask
+    ```
+    > Replace `xql-dask` with the name you want your dask cluster to have. Just set `DASK_CLUSTER={dask_cluster_name}` environment variable.
```
Comment (mahrsee1997) [inline xql/README.md:112]: nit: Where to replace ? Better to write the method name or line number.

File: xql/src/xql/utils.py [hunk 3]
```diff
@@ -26,3 +30,25 @@ def wrap(*args, **kw):
...
         return result
     return wrap
+
+
+def connect_dask_cluster() -> None:
+    """
+    Connects to a Dask cluster.
+    """
+    # Fetch the cluster name from environment variable, default to "era5-dask" if not set
+    cluster_name = os.getenv('DASK_CLUSTER', "xql-dask")
```
Comment (mahrsee1997) [inline xql/src/xql/utils.py:40]: nit: let's mention about this env variable in readme.

File: xql/src/xql/apply.py (hunk 1 above)
Comment (DarshanSP19) [reply xql/src/xql/apply.py]: Removed.

File: xql/README.md (hunk 2 above)
Comment (DarshanSP19) [reply xql/README.md]: It's just in the above command. Mentioned.

File: xql/src/xql/utils.py (hunk 3 above)
Comment (DarshanSP19) [reply xql/src/xql/utils.py:40]: Previous Comment.

File: xql/src/xql/utils.py [hunk 4]
```diff
@@ -26,3 +30,25 @@ def wrap(*args, **kw):
         print(f"Query took: { strftime('%H:%M:%S', gmtime(te - ts)) }")
         return result
     return wrap
+
+
+def connect_dask_cluster() -> None:
+    """
+    Connects to a Dask cluster.
+    """
+    # Fetch the cluster name from environment variable, default to "era5-dask" if not set
```
Comment (mahrsee1997) [inline xql/src/xql/utils.py:39]: nit: default to 'xql-dask' ?

**Review by mahrsee1997 — APPROVED** [review]
LGTM.

File: xql/src/xql/utils.py (hunk 4 above)
Comment (DarshanSP19) [reply xql/src/xql/utils.py]: Done.

**Pull request merged**